x-pack/metricbeat/module/aws/awshealth/awshealth.go

```diff
@@ -0,0 +1,314 @@
+	deInputParams := health.DescribeEventsInput{
+		Filter: &eventFilter,
+	}
+
+	// Define options for DescribeEventsPaginator
+	deOptions := &health.DescribeEventAggregatesPaginatorOptions{
+		Limit:                10,
+		StopOnDuplicateToken: true,
+	}
+
+	// Function option to apply options to the paginator
+	deOptFn := func(options *health.DescribeEventsPaginatorOptions) {
+		// Apply the provided options
+		options.Limit = deOptions.Limit
+		options.StopOnDuplicateToken = deOptions.StopOnDuplicateToken
+	}
+	// Define options for DescribeAffectedEntitiesPaginator
+	affOptions := &health.DescribeAffectedEntitiesPaginatorOptions{
+		Limit:                10,
+		StopOnDuplicateToken: true,
+	}
+	// Function option to apply options to the paginator
+	affOptFn := func(options *health.DescribeAffectedEntitiesPaginatorOptions) {
+		// Apply the provided options
+		options.Limit = affOptions.Limit
+		options.StopOnDuplicateToken = affOptions.StopOnDuplicateToken
+	}
+	// Create DescribeEventsPaginator with AWS Health client and options
+	dePage := health.NewDescribeEventsPaginator(awsHealth, &deInputParams, deOptFn)
+
+	for dePage.HasMorePages() {
+		healthDetailsTemp = []AWSHealthMetric{}
+
+		// Perform actions for the current page
+		currentPage, err := dePage.NextPage(ctx)
+		if err != nil {
+			m.Logger().Errorf("[AWS Health] DescribeEvents failed with : %w", err)
+			break
+		}
+		deEvents = currentPage.Events
+		eventArns := make([]string, len(deEvents))
+		// Iterate through events to extract relevant information
+		for i := range deEvents {
+			healthDetailsTemp = append(healthDetailsTemp, AWSHealthMetric{
+				EventArn:          awssdk.ToString(deEvents[i].Arn),
+				EndTime:           awssdk.ToTime(deEvents[i].EndTime),
+				EventScopeCode:    awssdk.ToString((*string)(&deEvents[i].EventScopeCode)),
+				EventTypeCategory: awssdk.ToString((*string)(&deEvents[i].EventTypeCategory)),
+				EventTypeCode:     awssdk.ToString(deEvents[i].EventTypeCode),
+				LastUpdatedTime:   awssdk.ToTime(deEvents[i].LastUpdatedTime),
+				Region:            awssdk.ToString(deEvents[i].Region),
+				Service:           awssdk.ToString(deEvents[i].Service),
+				StartTime:         awssdk.ToTime(deEvents[i].StartTime),
+				StatusCode:        awssdk.ToString((*string)(&deEvents[i].StatusCode)),
+			})
+			eventArns[i] = awssdk.ToString(deEvents[i].Arn)
+		}
+		// Fetch event details for the current page of events
+		eventDetails, err := awsHealth.DescribeEventDetails(ctx, &health.DescribeEventDetailsInput{
+			EventArns: eventArns,
+			Locale:    &locale,
+		})
+		if err != nil {
+			m.Logger().Errorf("[AWS Health] DescribeEventDetails failed with : %w", err)
+			break
+		}
+		// Fetch event description for the current page of events
+		successSet := eventDetails.SuccessfulSet
+		for x := range successSet {
+			for y := range healthDetailsTemp {
+				if awssdk.ToString(successSet[x].Event.Arn) == healthDetailsTemp[y].EventArn {
```

Comment:
we don't have to do this now, but perhaps for a future refactor, i wonder if it's simpler to maintain a map of ARNs to metric events,  instead of having these comparisons in multiple places.